### DISCUSSION TITLE
Test coverage of website features

### DISCUSSION THREAD

**Discussion by ctmbl** [iScsc/iscsc.fr#18] [Ideas]
Tests could be separated in different categories:
 - automatically/manually
 - at each PR/release
 - backend/frontend/both

So what features of the website should be tested:

Non-logged:
- [ ] read an article 
- [ ] redirect to home page
- [ ] sign up
- [ ] impossibility of deleting/creating articles
- [ ] ...

Logged:
- [ ] log in
- [ ] read an article
- [ ] create an article
- [ ] impossibility of deleting others articles
- [ ] ...

@atxr you may propose other features to test
@amtoine you may give an opinion on how to address these tests, how solid should they be (because a common risk is to make tests so stronger that you can't change anything

**Comment by amtoine** [comment]
mm :thinking: 

tbh, i do not even have an idea how to implement such tests :sweat_smile: 

a question for now: **what do you mean by "_logged_" and "_non-logged_"?** :confused:

**Reply by dolphinau** [reply]
I think it's tests as a logged user or not

**Reply by amtoine** [reply]
got it :+1: 
wanted to be sure :yum:

**Comment by dolphinau** [comment]
These one looks very important indeed!
I would add also test that must fail, for example create an article/delete when not logged in

**Reply by ctmbl** [reply]
That's right!

**Reply by amtoine** [reply]
could you add that to the list above @ctmbl? :yum:

**Reply by amtoine** [reply]
in the end, pretty much all the tests would be duplicated in "logged" and "non-logged" :ok_hand:
